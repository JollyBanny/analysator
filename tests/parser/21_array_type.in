var sum: integer;
    score: double;
    symbol: char;
    str: string;
    box: array [0..10, 0..10, 0..10] of integer;

begin
    writeln('Hello world');
end.