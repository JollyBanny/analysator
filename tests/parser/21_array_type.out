program
└──── program block
      ├──── VAR
      │     ├──── var declaration
      │     │     ├──── sum
      │     │     └──── integer
      │     ├──── var declaration
      │     │     ├──── score
      │     │     └──── double
      │     ├──── var declaration
      │     │     ├──── symbol
      │     │     └──── char
      │     ├──── var declaration
      │     │     ├──── str
      │     │     └──── STRING
      │     └──── var declaration
      │           ├──── box
      │           └──── ARRAY
      │                 ├──── ..
      │                 │     ├──── 0
      │                 │     └──── 10
      │                 └──── ARRAY
      │                       ├──── ..
      │                       │     ├──── 0
      │                       │     └──── 10
      │                       └──── ARRAY
      │                             ├──── ..
      │                             │     ├──── 0
      │                             │     └──── 10
      │                             └──── integer
      └──── Compound statement
            └──── Call statement
                  └──── writeln
                        └──── 'Hello world'
